app/src/org/commcare/android/database/SqlStorage.java

```diff
@@ -413,12 +449,13 @@ public boolean isEmpty() {
                 return getCoveringIndexIterator(db, minValue, maxValue, countValue);
             }
         }
-
-        String[] projection = includeData ? new String[]{DbUtil.ID_COL, DbUtil.DATA_COL} : new String[]{DbUtil.ID_COL};
-        Cursor c = db.query(table, projection, null, null, null, null, null);
-        return new SqlStorageIterator<>(c, this);
+        return null;
     }
 
+    protected Cursor getIterateCursor(SQLiteDatabase db, boolean includeData) {
+        String[] projection = includeData ? new String[]{DatabaseHelper.ID_COL, DatabaseHelper.DATA_COL} : new String[]{DatabaseHelper.ID_COL};
+        return db.query(table, projection, null, null, null, null, null);
+    }
```

Comment:
Split this logic into a helper method so subclass `FileBackedSqlStorage` can override it with appropriate logic.